cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -132,7 +120,7 @@ public final String toString() {
     for (String part : idParts) {
       result.append(part).append(IDSTRING_PART_SEPARATOR);
     }
-    if (!Iterables.isEmpty(idParts)) {
+    if (idParts.iterator().hasNext()) {
```

Comment:
Also, as a general rule of thumb, you can't expect two separate calls to `Iterable.iterator()` will give you the same sequence.